File: server/src/main/java/freelance/twin/sport/server/commande/entity/Commande.java
package freelance.twin.sport.server.commande.entity;

import freelance.twin.sport.server.user.entity.User;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "commandes", indexes = {
        @Index(name = "idx_commande_user_id", columnList = "user_id"),
        @Index(name = "idx_commande_status", columnList = "status"),
        @Index(name = "idx_commande_created_at", columnList = "created_at"),
        @Index(name = "idx_commande_ref", columnList = "ref")
})
public class Commande {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String ref;

    @Enumerated(EnumType.STRING)
    private Status status;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
    // ── Delivery ─────────────────────────────────────────────────────────
    @Enumerated(EnumType.STRING)
    private DeliveryMethod deliveryMethod;

    private String adress;
    private String city;
    private String phone;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = true)
    private User user;

    // ── Items ─────────────────────────────────────────────────────────────
    @OneToMany(mappedBy = "commande", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<CommandItem> items = new ArrayList<>();


    private double totalPrice;
    private double deliveryFee;

    private String notes;

    private String guestFirstName;
    private String guestLastName;
    private String guestEmail;
    private String guestPhone;
}

File: front/src/models/Commande.ts
// models/Commande.ts

import { CommandeItemDto, CommandeItemRequest, CommandItem } from "./CommandItem";
import { User } from "./User";

export enum DeliveryMethod {
  DELIVERY = "DELIVERY",
  PICKUP = "PICKUP",
}
export enum Status {
  EN_ATTENTE = "EN_ATTENTE",
  CONFIRMEE = "CONFIRMEE",
  EXPEDIEE = "EXPEDIEE",
  LIVREE = "LIVREE",
  PRETE_RETRAIT = "PRETE_RETRAIT",
  RECUPEREE = "RECUPEREE",
  ANNULEE = "ANNULEE",
}
export interface Commande {
  id: number;
  ref: string;
  status: Status;
  createdAt: string; 
  updatedAt?: string;

  // Delivery
  deliveryMethod: DeliveryMethod;
  adress?: string;
  city?: string;
  phone?: string;

  user?: User;

  // Items
  items: CommandItem[];

  totalPrice: number;
  deliveryFee: number;
  notes?: string;

  guestFirstName?: string;
  guestLastName?: string;
  guestEmail?: string;
  guestPhone?: string;
}
export interface CommandeFilterRequest {
  page: number;
  pageSize: number;
  userId: string | null;
  search: string | null;
  status: Status[] | null;
}


export interface CommandeDto {
  id: number;
  adress: string;
  city: string;
  ref: string;
  deliveryMethod: DeliveryMethod;
  notes: string;
  createdAt: string;
  status: Status;
  updatedAt: string;
  phone: string;
  totalPrice: number;
  deliveryFee: number;
  userId: string | null;
  username: string | null;
  userNom: string | null;
  userPrenom: string | null;
  guestFirstName: string | null;
  guestLastName: string | null;
  guestEmail: string | null;
  guestPhone: string | null;
  items: CommandeItemDto[];
}
export interface CommandeRequest {
  adress?: string;
  city?: string;
  ref?: string;
  deliveryMethod: DeliveryMethod;
  notes?: string;
  phone?: string;
  totalPrice?: number;
  deliveryFee?: number;
  guestFirstName?: string;
  guestLastName?: string;
  guestEmail?: string;
  guestPhone?: string;
  items?: CommandeItemRequest[];
}